pkg/build/admission/defaults/admission.go

```diff
@@ -171,3 +194,37 @@ func addDefaultLabel(defaultLabel buildapi.ImageLabel, buildLabels *[]buildapi.I
 		*buildLabels = append(*buildLabels, defaultLabel)
 	}
 }
+
+//setDefaultSourceSecret method check if sourcesecret is set via defaults options and return result or err in non found
+func (a *buildDefaults) setDefaultSourceSecret(build *buildapi.Build) (string, error) {
+	//check if project annotion is set for default SourceSecret
+	//second option is global config.
+	ns, err := a.cache.GetNamespace(build.Namespace)
+	if err == nil {
+		if contains(ns.Annotations, "openshift.io/sourceSecret") {
```

Comment:
is this so that project admins can customize the default secret for their namespace? normal project admins won't have write permission to these annotations